Sphere_3 intersection with interval arithmetic fails on 5.6 (works on 5.2.2)

Hi all,

Thanks for releasing CGAL 5.6.

The migration of our platform (sbl at sbl.inria.fr) from CGAL 5.5.2 to CGAL 5.6 goes pretty smoothly but for the following issue:
Intersection on the sphere with interval arithmetic feels partly broken.

## Issue Details

To exhibit the issue, I barely updated the sphere intersection example from the CGAL repository (https://github.com/CGAL/cgal/blob/5.6.x-branch/Circular_kernel_3/examples/Circular_kernel_3/intersecting_spheres.cpp).
This modified example (hereafter) compiles on 5.5.2, but not on 5.6.
The original example (with Exact_spherical_kernel) compiles both on 5.5.2 and 5.6.

I could not pinpoint the change triggering the regression. 
It may be that the snippet compilation on CGAL 5.5.2 was not even considered possible.

The compiler error is pretty straightforward : 
```
<>/CGAL/Circular_kernel_3/internal_functions_on_sphere_3.h:320:16: error:
error: no viable conversion from 'Needs_FT<result_type>' (aka 'Needs_FT<Uncertain<bool>>') to 'bool'
            if(SK().has_on_3_object()(s3, *c)) {
               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Obviously, changing CGAL's code to include 'CGAL::certainly', 'CGAL::certainly_not' wrapper around the has_on_3_object resulting 'Boolean' works (in both cases : exact and interval arithmetic).

I don't know if that will be what the fix will be about. 
Note : the compilation error only occurring for the Circle_3 case, and not Point_3. 

## Source Code
```c++
#include <CGAL/Simple_cartesian.h>
#include <CGAL/Spherical_kernel_3.h>
#include <CGAL/Exact_spherical_kernel_3.h>
#include <CGAL/Random.h>

typedef CGAL::Simple_cartesian<CGAL::Interval_nt<false> > Geometric_kernel_interval;
typedef CGAL::Algebraic_kernel_for_spheres_2_3<typename Geometric_kernel_interval::FT> Algebraic_kernel_interval;
typedef CGAL::Spherical_kernel_3<Geometric_kernel_interval, Algebraic_kernel_interval> Spherical_k; // -> compiles on 5.2.2, not on 5.6
// typedef CGAL::Exact_spherical_kernel_3         Spherical_k; // -> compiles on 5.5.2 and on 5.6

typedef CGAL::Point_3<Spherical_k>             Point_3;
typedef CGAL::Sphere_3<Spherical_k>            Sphere_3;

int main() {

  CGAL::Random generatorOfgenerator;
  int random_seed = generatorOfgenerator.get_int(0, 123456);
  CGAL::Random theRandom(random_seed);
  int count = 0;

  std::cout << "We will compute the approximate probability that 3 spheres wit"
  << "h radius 1 intersect on a 5x5x5 box, it might take some time." << std::endl;

  for(int i=0; i<10000; i++) {

    double x1 = theRandom.get_double(0.0,5.0);
    double y1 = theRandom.get_double(0.0,5.0);
    double z1 = theRandom.get_double(0.0,5.0);
    double r = 1.0;
    double x2 = theRandom.get_double(0.0,5.0);
    double y2 = theRandom.get_double(0.0,5.0);
    double z2 = theRandom.get_double(0.0,5.0);
    double x3 = theRandom.get_double(0.0,5.0);
    double y3 = theRandom.get_double(0.0,5.0);
    double z3 = theRandom.get_double(0.0,5.0);

    Sphere_3 s1 = Sphere_3(Point_3(x1,y1,z1), r);
    Sphere_3 s2 = Sphere_3(Point_3(x2,y2,z2), r);
    Sphere_3 s3 = Sphere_3(Point_3(x3,y3,z3), r);

    std::vector< CGAL::Object > intersecs;
    typename Spherical_k::Intersect_3 intersection_test;
    intersection_test(s1, s2, s3, std::back_inserter(intersecs));
    if(intersecs.size() > 0) count++;
  }

  std::cout << "The approximate probability that 3 spheres with radius 1"
            << std::endl;
  std::cout << "chosen (uniformly) randomly on a 5x5x5 box intersect is: "
            << (static_cast<double>(count))/(static_cast<double>(10000)) << std::endl;

  return 0;
}
```

## Environment

* Operating system (Windows/Mac/Linux, 32/64 bits): Mac M1 arm64 
* Compiler: clang version 16.0.3
* Release or debug mode: Debug
* Specific flags used (if any): 
    * CXX_FLAGS                  -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -stdlib=libc++ -fvisibility-inlines-hidden -fmessage-length=0 -isystem /path_to_own_project_include_dirs
     * CXX_FLAGS_DEBUG            -g
* CGAL version: compare 5.6 to 5.5.2 (conda install)

--- a/Circular_kernel_3/include/CGAL/Circular_kernel_3/internal_functions_on_sphere_3.h
+++ b/Circular_kernel_3/include/CGAL/Circular_kernel_3/internal_functions_on_sphere_3.h
@@ -278,6 +278,7 @@ namespace CGAL {
        typedef typename SK::Point_3  Point_3;
        typedef typename SK::Sphere_3  Sphere_3;
        typedef typename SK::Algebraic_kernel  Algebraic_kernel;
+       typedef typename SK::Boolean Bool;
        typedef typename SK3_Intersection_traits<SK, Sphere_3, Sphere_3, Sphere_3>::type result_type;
 
        CGAL_kernel_precondition(!s1.is_degenerate());
@@ -317,7 +318,7 @@ namespace CGAL {
              return res;
          }
          if(const Circle_3* c = CGAL::Intersections::internal::intersect_get<Circle_3>(v)) {
-            if(SK().has_on_3_object()(s3, *c)) {
+            if(static_cast<Bool>(SK().has_on_3_object()(s3, *c))) {
               *res++ = result_type(*c);
             }
            return res;

